packages/relay/src/lib/eth.ts

```diff
@@ -1633,6 +1646,47 @@ export class EthImpl implements Eth {
     }
   }
 
+  /**
+   * Sends a raw transaction with retry logic.
+   *
+   * @template T The type of the result returned by the sendRawTransaction function.
+   * @param {Function} sendRawTransaction The function responsible for sending the raw transaction.
+   * @param {object} options The options for retrying the transaction.
+   * @param {number} [options.maxAttempts=2] The maximum number of attempts to send the transaction.
+   * @param {number} [options.backOff=500] The backoff period in milliseconds between retry attempts.
+   * @param {(error: unknown) => boolean} [options.canRetry=(error) => true] A function that determines whether a retry attempt can be made based on the error received.
+   * @param {(error: SDKClientError) => void} [options.onError=(error) => {}] A function to handle errors that occur during retry attempts.
+   * @returns {Promise<T>} A promise resolving to the result of the transaction.
+   */
+  private async sendRawTransactionWithRetry<T>(
+    sendRawTransaction: () => T,
+    {
+      maxAttempts = 2,
+      backOff = 500,
+      canRetry = (e: unknown) => true,
+      onError = (e: SDKClientError) => {},
+    }: {
+      maxAttempts?: number;
+      backOff?: number;
+      canRetry?: (error: unknown) => boolean;
+      onError?: (error: SDKClientError) => void;
+    },
+  ): Promise<T | undefined> {
```

Comment:
Given the function returns either a transaction or throws an error, is the `undefined` generic necessary?

Reply:
Well the reason is that the return statement is in the try, so if this fails we throw an error, but do not return anything and in this case typescript is warning us that it is possible for the return type to be undefined, thus we add this. Just trying to make TS happy :)

Reply:
Okay.